src/makeInfinityAdder.js

```diff
@@ -1,10 +1,27 @@
 'use strict';
 
 /***
+ *
  * @return {function}
  */
+
 function makeInfinityAdder() {
-  // write code here
+  let result = 0;
+
+  const adder = (arg) => {
+    if (arg === undefined) {
+      const sum = result;
+      result = 0;
+
+      return sum;
+    }
+
+    result += arg;
+
+    return adder;
+  };
```

Comment:
Consider adding a default value for the `arg` parameter in the `adder` function. This would make the check against `undefined` more explicit and align with modern JavaScript practices of using default parameters. For example, `const adder = (arg = undefined) => {...}`.